tests/test_tokenization_barthez.py

```diff
@@ -0,0 +1,78 @@
+# coding=utf-8
+# Copyright 2020 Ecole Polytechnique and HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import unittest
+
+from transformers import BarthezTokenizer, BatchEncoding
+from transformers.testing_utils import require_sentencepiece, require_tokenizers, require_torch
+
+from .test_tokenization_common import TokenizerTesterMixin
+
+
+@require_tokenizers
+@require_sentencepiece
+class BarthezTokenizationTest(TokenizerTesterMixin, unittest.TestCase):
+
+    tokenizer_class = BarthezTokenizer
+    test_rust_tokenizer = False
```

Comment:
Thanks for implementing the fast tokenizer! It would be nice to add it to the tests.

Could you switch that to `True` and test the fast tokenizer as well? Thanks!

Reply:
Thank you @LysandreJik, I switched to the fast tokenization test. However I noticed that if the sentencepiece model uses bpe tokenization instead of unigram (model_type = 2) https://github.com/huggingface/transformers/blob/367f497dec8905ac37c2947be998f4469fdade6b/src/transformers/convert_slow_tokenizer.py#L309)
the conversion to fast tokenizer  will be slow, and the integration test will be slow as well. In my case BARThez model uses type 2, do you think it is possible to save the `tokenizer.json` when calling  `tokenizer.save_pretrained()` to avoid that the model performs the conversion when loading the tokenizer from a saved directory?

Reply:
Indeed, that takes a while. Let me have a look.

Reply:
You can use `legacy_format=False` to save the `tokenizer.json` file directly, but I feel like this should be automatically done for the fast tokenizers. Looking into it.